### DISCUSSION TITLE
ACP-20: Ed25519 p2p

### DISCUSSION THREAD

**Discussion by avalanche-foundation-admin** [avalanche-foundation/ACPs#21] [Discussions]
Support Ed25519 TLS certificates for p2p communications on the Avalanche network. Permit usage of Ed25519 public keys for Avalanche Network Client (ANC) NodeIDs. Support Ed25519 signatures in the ProposerVM.

https://github.com/avalanche-foundation/ACPs/blob/main/ACPs/20-ed25519-p2p/README.md

**Comment by patrick-ogrady** [comment]
Would it make sense to find a way to use BLS keys for this instead?

Seems nice to only have 1 secret for validators to worry about (and seems like we'll use BLS for the foreseeable future because of Avalanche Warp Messaging).

**Reply by mtgnoah** [reply]
@patrick-ogrady Would that still use TLS Certificates or move entirely to BLS? A move to BLS could make sense because it would allow Avalanche to take advantage of the large adoption of BLS as seen in it's usage in Ethereum post-Merge.

**Reply by dhrubabasu** [reply]
@mtgnoah This would not use TLS certificates as TLS certificates do not support BLS.

I'm not sure if there are any authenticated communication protocols that support BLS: they're slow to verify and not really suitable _unless_ you are aggregating signatures.

Decreasing the amount of things validators need to manage is a great goal. The ACP achieves this by reducing the number from 3 (BLS key, RSA key, TLS certificate) to 2 (BLS key, Ed25519 key).

**Comment by softstackio** [comment]
How far is this discussion? Cause most institutional custody provider would prefer support for Ed25519.
